packages/sit-onyx/src/components/OnyxSelect/types.ts

```diff
@@ -17,10 +17,17 @@ export type SelectSearchProps =
        * Property is managed internally, when undefined.
        */
       searchTerm?: string;
+      /**
+       * A filtered subset of the `options`.
+       * As default, onyx will handle the search by comparing
+       * the option labels with the `searchTerm`.
+       */
+      filteredOptions?: SelectOption<TValue>[];
```

Comment:
Let's sync about the API here. When we are on the same page here, we can ensure a more stable API :) @MajaZarkova @larsrickert @BoppLi Do you have time today?
```
      withSearch: true | {
        /**
         * A filter function for synchronous filtering of the provided `options`. 
         * The default filter function will compare
         * the option labels with the `searchTerm`.
         */
        filterFunc: (option: SelectOption<TValue>, index: number, array: SelectOption<TValue>[]) => boolean;
      } | {
        loading?: boolean;
      }
```

Reply:
note: as discussed, for this PR this would be too much. implemented another approach removing the filteredOptions for now